src/java/org/apache/cassandra/tools/nodetool/DataPaths.java

```diff
@@ -38,16 +45,46 @@ public class DataPaths extends NodeToolCmd
     @Option(title = "format", name = {"-F", "--format"}, description = "Output format (json, yaml)")
     private String outputFormat = "";
 
+    @Option(title = "",
+            name = {"--list-orphaned"},
+            description = "List directories in the data paths which are not controlled by Cassandra")
+    private Boolean orphaned = false;
+
     @Override
     protected void execute(NodeProbe probe)
     {
         if (!outputFormat.isEmpty() && !"json".equals(outputFormat) && !"yaml".equals(outputFormat))
-        {
             throw new IllegalArgumentException("arguments for -F are yaml and json only.");
-        }
 
         DataPathsHolder holder = new DataPathsHolder(probe, tableNames);
-        StatsPrinter<DataPathsHolder> printer = DataPathsPrinter.from(outputFormat);
-        printer.print(holder, probe.output().out);
+        if (orphaned)
+        {
+            if (!outputFormat.isEmpty() || !tableNames.isEmpty())
+                throw new IllegalArgumentException("Format option, keyspace and table name arguments aren't supported when listing orphans.");
+
+            Set<String> dataDirs = new HashSet<>();
+            dataDirs.addAll(Arrays.asList(probe.getStorageService().getNonLocalSystemKeyspacesDataFileLocations()));
+            dataDirs.addAll(Arrays.asList(probe.getStorageService().getLocalSystemKeyspacesDataFileLocations()));
+            
+            Set<File> allFiles = new HashSet<>();
```

Comment:
If you want to use `File`, we have a special `File` class in `org.apache.cassandra.io.util.File` instead of `java.io.File`. I think it would fail to compile by ant if you used the original Java one, there is some check for it.